packages/server/lib/browsers/firefox.ts

```diff
@@ -23,6 +23,13 @@ import type { RemoteConfig } from 'webdriver'
 import type { GeckodriverParameters } from 'geckodriver'
 import { WebDriver } from './webdriver'
 
+export class CDPFailedToStartFirefox extends Error {
+  constructor (message) {
+    super(message)
+    this.name = 'CDPFailedToStartFirefox'
```

Comment:
I recommend a custom error type pattern I use elsewhere:


```typescript
const ErrorKind = 'SomeError'

class SomeError extends Error {
  public readonly kind = ErrorKind
  public static isSomeError (error: Error & { kind?: any }): error is SomeError {
    return error.kind === ErrorKind
  }
}
```

This allows for type inference along with logic related to error handling, e.g. 

```typescript
try {
  // etc
} catch (e) {
  if (SomeError.isSomeError(e)) {
    // typescript infers that e is of type `SomeError`
  }
}
```

Reply:
how does 0a0bcf9bf131232dc6e2be3e1fb652339f0beaea look?